test/e2e/ingress.go

```diff
@@ -135,6 +143,32 @@ var _ = framework.KubeDescribe("Loadbalancing: L7 [Feature:Ingress]", func() {
 			By("should reject HTTP traffic")
 			ExpectNoError(pollURL(fmt.Sprintf("http://%v/", ip), "", lbPollTimeout, httpClient, true))
 
+			By("should have correct firewall rule for ingress")
+			fw, err := getFwForIngress(cloudConfig)
+			Expect(err).NotTo(HaveOccurred())
+			// Retrieve nodeTags from the node instance as the target tags for the ingress firewall rule
+			nodes := framework.GetReadySchedulableNodesOrDie(jig.client)
+			Expect(len(nodes.Items) > 0).Should(BeTrue())
+			nodeTags, err := getInstanceTags(cloudConfig, nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+			// Getting ingress backend nodePorts. Current GCE ingress controller allows
+			// traffic to the default backend nodePort by default, so retrieve that as well.
+			nodePorts := []string{}
+			defaultSvc, err := jig.client.Core().Services(api.NamespaceSystem).Get(defaultBackendName)
```

Comment:
can you please make this a method on the jig called `getIngressNodePorts(ingName, ns)` that does the following:
1. get the ingress
2. get its services:service-port
3. get each service and find the nodeport associated with service-port
4. return a list of int nodeports

you can just have this method Failf internally instead of returning an error, keeps the main test cleaner and easier to reason about